File: src/main/java/aurocosh/divinefavor/common/potions/potions/PotionCrushingPalm.java
package aurocosh.divinefavor.common.potions.potions;

import aurocosh.divinefavor.common.custom_data.player.PlayerData;
import aurocosh.divinefavor.common.custom_data.player.data.talisman_uses.TalismanUsesData;
import aurocosh.divinefavor.common.item.talismans.spell.base.ItemSpellTalisman;
import aurocosh.divinefavor.common.network.message.client.spell_uses.MessageSyncSpellUses;
import aurocosh.divinefavor.common.potions.base.potion.ModPotionToggleLimited;
import aurocosh.divinefavor.common.potions.common.ModPotions;
import aurocosh.divinefavor.common.util.UtilBlock;
import net.minecraft.block.Block;
import net.minecraft.block.state.IBlockState;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.item.ItemStack;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.World;
import net.minecraftforge.event.entity.player.PlayerInteractEvent;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

@Mod.EventBusSubscriber
public class PotionCrushingPalm extends ModPotionToggleLimited {

    public PotionCrushingPalm() {
        super("crushing_palm", true, 0x7FB8A4);
    }

    @SubscribeEvent
    public static void onPlayerLeftClickBlock(PlayerInteractEvent.LeftClickBlock event) {
        World world = event.getWorld();
        if (world.isRemote)
            return;

        EntityPlayer player = event.getEntityPlayer();
        if (!player.isPotionActive(ModPotions.crushing_palm))
            return;

        BlockPos pos = event.getPos();
        IBlockState state = world.getBlockState(pos);
        Block block = state.getBlock();
        if (!block.isToolEffective("pickaxe", state))
            return;

        TalismanUsesData usesData = PlayerData.get(player).getTalismanUsesData();
        ItemSpellTalisman talisman = ModPotions.crushing_palm.getTalisman();
        if (!usesData.consumeUse(talisman.getId()))
            return;

        new MessageSyncSpellUses(talisman.getId(), usesData).sendTo(player);

        ItemStack stack = player.getHeldItemMainhand();
        UtilBlock.removeBlockWithDrops(player, world, stack, pos, false, true);
    }

    @Override
    public boolean isReady(int duration, int amplifier) {
        return false;
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/potions/potions/PotionCrushingPalm.kt
package aurocosh.divinefavor.common.potions.potions

import aurocosh.divinefavor.DivineFavor
import aurocosh.divinefavor.common.item.spell_talismans.context.CastContextGenerator
import aurocosh.divinefavor.common.potions.base.potion.ModPotionToggleLimited
import aurocosh.divinefavor.common.potions.common.ModPotions
import aurocosh.divinefavor.common.util.UtilBlock
import net.minecraft.item.ItemStack
import net.minecraftforge.event.entity.player.PlayerInteractEvent
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent

class PotionCrushingPalm : ModPotionToggleLimited("crushing_palm", 0x7FB8A4) {
    override fun isReady(duration: Int, amplifier: Int): Boolean {
        return false
    }
}

@Mod.EventBusSubscriber(modid = DivineFavor.MOD_ID)
object CrushingPalm {
    @SubscribeEvent
    @JvmStatic
    fun onPlayerLeftClickBlock(event: PlayerInteractEvent.LeftClickBlock) {
        val world = event.world
        if (world.isRemote)
            return

        val player = event.entityPlayer
        if (!player.isPotionActive(ModPotions.crushing_palm))
            return

        val talisman = ModPotions.crushing_palm.talisman
        val context = CastContextGenerator.player(player)
        if (!talisman.claimCost(context))
            return

        val pos = event.pos
        val state = world.getBlockState(pos)
        val block = state.block
        if (block.isToolEffective("pickaxe", state))
            UtilBlock.removeBlock(player, world, ItemStack.EMPTY, pos, true, false, true)
    }
}